media-libs/libjxl/libjxl-0.7.0_pre20220825.ebuild

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake-multilib
+
+DESCRIPTION="JPEG XL image format reference implementation"
+HOMEPAGE="https://github.com/libjxl/libjxl"
+
+SRC_URI="https://api.github.com/repos/libjxl/libjxl/tarball/4c31ef03e3fd5239d6b794771d4ae7daa7815b28 -> ${P}.tar.gz"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~riscv ~sparc ~x86"
+
+LICENSE="BSD"
+SLOT="0"
+IUSE="openexr"
+
+DEPEND="app-arch/brotli:=[${MULTILIB_USEDEP}]
+	dev-cpp/gflags:=[${MULTILIB_USEDEP}]
+	>=dev-cpp/highway-1.0.0[${MULTILIB_USEDEP}]
+	media-libs/giflib:=[${MULTILIB_USEDEP}]
+	>=media-libs/lcms-2.13:=[${MULTILIB_USEDEP}]
+	media-libs/libpng:=[${MULTILIB_USEDEP}]
+	sys-libs/zlib[${MULTILIB_USEDEP}]
+	media-libs/libjpeg-turbo[${MULTILIB_USEDEP}]
+	>=x11-misc/shared-mime-info-2.2
+	openexr? ( media-libs/openexr:= )
+"
+RDEPEND="${DEPEND}"
+
+S="${WORKDIR}/libjxl-libjxl-4c31ef0"
```

Comment:
Move `S` to just after `SRC_URI`. While at it, maybe set `COMMIT=4c31ef03e3fd5239d6b794771d4ae7daa7815b28` so you can use it in `SRC_URI` then again in `S` as `${COMMIT:0:7}`?